### PR TITLE
Update upgrade script for 1.4.1

### LINKED ISSUE #1806: Support appliance 1.4.1 Upgrade
##### For feature requests/user stories, please include the information below:

**User Statement:**

As a vSphere admin I'd like to upgrade the VIC 1.4.0 ova to VIC 1.4.1.

**Details:**

The upgrade script needs to be edited to support the vic 1.4.0 tag as a recognized upgrade path.

**Acceptance Criteria:**
- [ ] Upgrade 1.4.0 to 1.4.1 is a recognized path
- [ ] Nightly upgrade test added for 1.4.0 to 1.4.1

### PR DESCRIPTION
**Pull request by morris-jason**
VIC Appliance Checklist:
- [x] Up to date with `master` branch
- [ ] Added tests
- [ ] Considered impact to upgrade
- [x] Tests passing
- [ ] Updated documentation
- [ ] Impact assessment checklist

If this is a feature or change to existing functionality, consider areas of impact with the [Impact
Assessment Checklist](https://github.com/vmware/vic-product/blob/master/installer/docs/CHANGE.md)

Fixes #1806.

Adds Manual and Automated Nightly upgrade tests for ova upgrade.
Adds 1.4.0 as a valid upgrade version for v1.4.1.

Request for 1.4.1 cherry pick.

### CODE REVIEW

File: tests/manual-test-cases/Group7-Upgrade/7-04-Upgrade-1.4.0.robot [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+*** Settings ***
+Documentation  Test 7-04 - Upgrade 1.4.0
+Resource  ../../resources/Util.robot
+Suite Setup  Wait Until Keyword Succeeds  10x  10m  OVA Upgrade Setup
```
Comment (zjs) [inline tests/manual-test-cases/Group7-Upgrade/7-04-Upgrade-1.4.0.robot]: Might  it be better to reorganize the suites in this group into a single suite, with one test case for each version?

Having 4 single-test suites, each of which deploy a whole VC environment and use it for a single simple test case, seems... wasteful.

Alternatively, is there some pattern for share the VC environment between suites?

(This may be better addressed as a separate PR.)

Comment (mdharamadas1) [reply tests/manual-test-cases/Group7-Upgrade/7-04-Upgrade-1.4.0.robot]: I think separate VCs for each of these tests is intentional design to keep the environments isolated and also as we run these tests concurrently, suite is much faster this way.
